examples/dreamcast/basic/watchdog/watchdog.c

```diff
@@ -0,0 +1,161 @@
+/* KallistiOS ##version##
+
+   watchdog.c
+   Copyright (C) 2023 Falco Girgis
+
+*/
+
+/* This program serves as both an example of using the Watchdog Timer
+   peripheral on the Dreamcast's SH4 CPU as well as a test to validate
+   the behavior of its driver in KOS.
+
+   NOTE:
+   At the time this is being written, there are no Dreamcast emulators
+   out there which are bothering to emulate the SH4 WDT, since apparently
+   no commercial games actually used it. Special care has been taken to
+   fail the tests gracefully when there is no functioning WDT.
+*/ 
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdatomic.h>
+#include <time.h>
+
+#include <arch/wdt.h>
+#include <dc/maple/controller.h>
+
+/* Microsecond-based units */
+#define MSEC            1000
+#define SEC             (1000 * MSEC)
+
+/* Test configuration constants */
+#define WDT_PET_COUNT   2000
+#define WDT_INTERVAL    (500 * MSEC)
+#define WDT_SECONDS     10 
+#define WDT_COUNT_MAX   ((WDT_SECONDS * SEC) / WDT_INTERVAL)
+
+/* User callback to be invoked from the WDT's interval timer 
+   interrupt when the requested microsecond period has elapsed. 
+   Our generic userdata pointer is passed back to us as void*. 
+   We use it to store our atomic counter to count the number 
+   of times this callback has been invoked. 
+
+   NOTE: Since we're accessing both from within and outside
+   of the interrupt, we make it atomic to ensure that it 
+   doesn't get updated while it's being read or vice-versa. */
+static void wdt_timeout(void *user_data) {
+    ++(*(atomic_uint *)user_data);
+}
+
+/* Main entry point */
+int main(int argc, char *argv[]) { 
+    bool success = true;
+
+    /* Press all buttons simultaneously to exit early. */
+    cont_btn_callback(0, CONT_START | CONT_A | CONT_B | CONT_X | CONT_Y,
+                      (cont_btn_callback_t)exit);
+
+    /* Note that is EXTREMELY important that the WDT gets disabled 
+       when it is done being used, otherwise it can continue running
+       after the application exits and can interfere with DC-Load 
+       operation. Since we have multiple exit points (one from returning
+       from main and one from the button callback), the safest thing to do
+       is to register wdt_disable() to be called automatically upon exit. */
+    atexit(wdt_disable);
+
+    printf("\nEnabling WDT in watchdog mode!\n");
+
+    /* Enable watchdog mode with a period of 5.25ms, causing a manual 
+       reset interrupt signal to be raised upon timeout. This will 
+       case your Dreamcast to reboot immediately. */
```

Comment:
```suggestion
       cause your Dreamcast to reboot immediately. */
```

Reply:
Damn. I swear I even reread all of the comments at least once... Thank you SO MUCH for taking the time to find the typos!